tests/unit/test_tm1.py

```diff
@@ -0,0 +1,111 @@
+from unittest.mock import patch
+
+import pandas as pd
+import pytest
+
+from viadot.exceptions import ValidationError
+from viadot.sources import TM1
+from viadot.sources.tm1 import TM1Credentials
+
+
+data = {"a": [420, 380, 390], "b": [50, 40, 45]}
+
+
+@pytest.fixture
+def tm1_credentials():
+    return TM1Credentials(
+        username="test_user",
+        password="test_password",  # pragma: allowlist secret # noqa: S106
+        address="localhost",
+        port="12356",
+    )
+
+
+@pytest.fixture
+def tm1(tm1_credentials: TM1Credentials):
+    return TM1(
+        credentials={
+            "username": tm1_credentials.username,
+            "password": tm1_credentials.password,
+            "address": tm1_credentials.address,
+            "port": tm1_credentials.port,
+        },
```

Comment:
You can use a shorthand
```suggestion
        credentials=tm1_credentials.dict(),
```